iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers.java

```diff
@@ -0,0 +1,344 @@
+        TwoWindingsTransformer t2wLeg2 = createTwoWindingsTransformer(t3w, t3w.getLeg2(), "Leg2", starVoltageLevel, getConnectivityStar(2, starVoltageLevel));
+        TwoWindingsTransformer t2wLeg3 = createTwoWindingsTransformer(t3w, t3w.getLeg3(), "Leg3", starVoltageLevel, getConnectivityStar(3, starVoltageLevel));
+
+        controlledRegulatingTerminals.replaceRegulatingTerminal(t3w.getLeg1().getTerminal(), t2wLeg1.getTerminal1());
+        controlledRegulatingTerminals.replaceRegulatingTerminal(t3w.getLeg2().getTerminal(), t2wLeg2.getTerminal1());
+        controlledRegulatingTerminals.replaceRegulatingTerminal(t3w.getLeg3().getTerminal(), t2wLeg3.getTerminal1());
+
+        // t2wLeg1, t2wLeg, and t2wLeg3 are not considered in controlledRegulatingTerminals (created in the model later)
+        Map<Terminal, Terminal> regulatingTerminalMapping = getRegulatingTerminalMapping(t3w, t2wLeg1, t2wLeg2, t2wLeg3);
+        replaceRegulatingTerminal(t2wLeg1, regulatingTerminalMapping);
+        replaceRegulatingTerminal(t2wLeg2, regulatingTerminalMapping);
+        replaceRegulatingTerminal(t2wLeg3, regulatingTerminalMapping);
+
+        copyTerminalActiveAndReactivePower(t3w.getLeg1().getTerminal(), t2wLeg1.getTerminal1());
+        copyTerminalActiveAndReactivePower(t3w.getLeg2().getTerminal(), t2wLeg2.getTerminal1());
+        copyTerminalActiveAndReactivePower(t3w.getLeg3().getTerminal(), t2wLeg3.getTerminal1());
+
+        List<String> lostProperties = copyProperties(t3w, t2wLeg1, t2wLeg2, t2wLeg3, starVoltageLevel);
+        List<String> lostExtensions = copyExtensions(t3w, t2wLeg1, t2wLeg2, t2wLeg3);
+
+        // copy necessary data before removing the transformer
+        String t3wId = t3w.getId();
+        List<AliasR> t3wAliases = getAliases(t3w);
+        t3w.remove();
+
+        // after removing the threeWindingsTransformer
+        List<AliasR> lostAliases = copyAliases(t3wAliases, t2wLeg1, t2wLeg2, t2wLeg3);
+
+        // warnings
+        if (!lostProperties.isEmpty()) {
+            lostProperties.forEach(propertyName -> logOrThrow(throwException, THREE_WINDINGS_TRANSFORMER + "'" + t3wId + "' unexpected property '" + propertyName + "'"));
+        }
+        if (!lostExtensions.isEmpty()) {
+            lostExtensions.forEach(extensionName -> logOrThrow(throwException, THREE_WINDINGS_TRANSFORMER + "'" + t3wId + "' unexpected extension '" + extensionName + '"'));
+        }
+        if (!lostAliases.isEmpty()) {
+            lostAliases.forEach(aliasR -> logOrThrow(throwException, THREE_WINDINGS_TRANSFORMER + "'" + t3wId + "' unexpected alias '" + aliasR.alias + "' '" + aliasR.aliasType + "'"));
+        }
+
+        // report
+        ReportNode reportNodeReplacement = reportNode.newReportNode().withMessageTemplate("replaced-t3w-by-3t2w", "Replaced ThreeWindingsTransformer by 3 TwoWindingsTransformers").add();
```

Comment:
I moved the report code to a private method in the same class to avoid moving the record definitions to the ModificationReports class. Let me know If you still prefer moving all this code to the ModificationReports class.

Reply:
I meant only the following piece:
```java
ReportNode reportNodeReplacement = reportNode.newReportNode().withMessageTemplate("replaced-t3w-by-3t2w", "Replaced ThreeWindingsTransformer by 3 TwoWindingsTransformers").add();
```
This should be moved to the ModificationReports class in case someone needs to check the existing report key/message (to reuse them or to translate them).

Reply:
Done

Reply:
Thank you for moving the newReportNode to the ModificationReports class. Could you add line breaks though to make it more consistent with the rest of the report methods and add a severity (Info I guess)?

``` java
return reportNode.newReportNode()
                .withMessageTemplate("replaced-t3w-by-3t2w", "Replaced ThreeWindingsTransformer by 3 TwoWindingsTransformers")
                .withSeverity(TypedValue.INFO_SEVERITY)
                .add();
```

thanks!

Reply:
Done